addons/website_crm_partner_assign/controllers/main.py

```diff
@@ -227,23 +227,15 @@ def sitemap_partners(env, rule, qs):
             if not qs or qs.lower() in loc:
                 yield {'loc': loc}
 
-    @http.route([
-        '/partners',
-        '/partners/page/<int:page>',
-
-        '/partners/grade/<model("res.partner.grade"):grade>',
-        '/partners/grade/<model("res.partner.grade"):grade>/page/<int:page>',
-
-        '/partners/country/<model("res.country"):country>',
-        '/partners/country/<model("res.country"):country>/page/<int:page>',
-
-        '/partners/grade/<model("res.partner.grade"):grade>/country/<model("res.country"):country>',
-        '/partners/grade/<model("res.partner.grade"):grade>/country/<model("res.country"):country>/page/<int:page>',
-    ], type='http', auth="public", website=True, sitemap=sitemap_partners, readonly=True)
-    def partners(self, country=None, grade=None, page=0, **post):
+    def _get_partners_values(self, country=None, grade=None, page=0, references_per_page=20, **post):
         country_all = post.pop('country_all', False)
         partner_obj = request.env['res.partner']
         country_obj = request.env['res.country']
+
+        industries = request.env['res.partner.industry'].sudo().search([])
+        industry_param = request.env['ir.http']._unslug(post.pop('industry', ''))[1]
```

Comment:
Not a big fan of `pop`, but I guess you need it to avoid issues in `searches` and `search_path`, right?